proposals/nnnn-main-menu-updating.md

```diff
@@ -0,0 +1,182 @@
+    </param>
+
+    <param name="updateSubCells" type="Bool" mandatory="false">
+        <description>If not set, assume false. If true, the app should send AddCommands with parentIDs matching the menuID. These AddCommands will then be attached to the submenu and displayed if the submenu is selected.</description>
+    </param>
+</function>
+
+<function name="UI.UpdateSubMenu" messagetype="response">
+</function>
+```
+
+#### Modifications
+```xml
+<function name="UI.AddCommand" messagetype="request">
+    <!-- New Parameters -->
+    <param name="hasIcon" type="Bool" mandatory="false" since="x.x">
+        <description>If not set, assume false. If set, then the AddCommand has an icon that has not been set in menuIcon but can be set if the module requests with UpdateCommand.</description>
+    </param>
+</function>
+
+<function name="UI.AddSubMenu" messagetype="request">
+    <!-- New Parameters -->
+    <param name="hasIcon" type="Bool" mandatory="false" since="x.x">
+        <description>If not set, assume false. If set, then the AddSubMenu has an icon that has not been set in menuIcon but can be set if the module requests with UpdateSubMenu.</description>
+    </param>
+</function>
+```
+
+### Manager-Level Updates
+The app library Menu Manager will need updates to work properly with this system. The menu manager will have to know when dynamic menu features are available and send only the appropriate data. 
+
+When `WindowCapability.supportsDynamicMenus` is `true` the menu managers will not send artworks until `UpdateCommand` or `UpdateSubMenu` is called with `UpdateArtwork`. At that point, it will have to upload the artwork and once that's finished, send a new `AddCommand` or `AddSubMenu` with the file name in the `image` parameter. 
+
+It will also not send sub-menus until `UpdateSubMenu` is called with `UpdateSubCells` as `true`. At that point, it will have to send `AddCommand` for all the submenu cells (while still respecting).
+
+However, I do not believe any new APIs will need to be added to the manager system, and therefore this does not need to be designed in this proposal.
+
+## Potential downsides
+This adds complexity to the HMI, which will have to keep track of which menu items exist, have artwork available but not on the system, and so forth. It will need to determine when it wants to send the `UpdateXXX` notifications. e.g. Should it send when the cell / sub-menu is on screen, or at some point before? This is a moderate complexity increase. The internal database of which cells need artwork is not too difficult due to the `AddCommand` / `AddSubMenu` not setting an image name until the artwork is actually uploaded.
```

Comment:
done

Reply:
Please update 
for example, should it send when the cell / sub-menu is on screen, or at some point before (this is a moderate complexity increase)?

to

For example, should it send when the cell / sub-menu is on screen, or at some point before? (this would be a moderate complexity increase).